vignettes/adrs.Rmd

```diff
@@ -422,34 +455,35 @@ dataset_vignette(
 
 ## Derive Clinical Benefit Parameter {#cb}
 
-Similarly, we now define the source location for this newly derived
-first response date.
+The function `derive_extreme_event()` can then be used to derive the clinical
+benefit parameter, which we define as a patient having had a response or a
+sustained period of time before first `PD`. This could also be known as disease
+control. In this example the "sustained period" has been defined as 42 days
+after randomization date via the `cb_y`[^2] event.
 
+[^2]:
 ```{r}
-resp <- date_source(
-  dataset_name = "adrs",
-  date = ADT,
-  filter = PARAMCD == "RSP" & AVALC == "Y"
-)
+cb_y
 ```
-
-The function `derive_param_clinbenefit()` can then be used to derive the
-clinical benefit parameter, which we define as a patient having had a
-response or a sustained period of time before first `PD`. This could
-also be known as disease control. In this example the "sustained period"
-has been defined as 42 days after randomization date, using the
-`ref_start_window` argument.
-
 ```{r}
 adrs <- adrs %>%
-  derive_param_clinbenefit(
-    dataset_adsl = adsl,
-    filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
-    source_resp = resp,
-    source_pd = pd,
-    source_datasets = list(adrs = adrs),
-    reference_date = RANDDT,
-    ref_start_window = 42,
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT, PARAMCD),
+    mode = "first",
+    events = list(
+      event(
+        condition = PARAMCD == "RSP" & AVALC == "Y",
+        set_values_to = exprs(AVALC = "Y")
+      ),
+      cb_y,
+      no_data_n
+    ),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    ignore_event_order = TRUE,
```

Comment:
could we add an explanation of why this is set `TRUE` here whereas it wasn't in the above derived parameter? reading the admiral PR i'm trying to make sense of this argument and my best guess is that we want the earliest of either the response or the sustained benefit (i.e. it doesn't matter which), whereas in BOR it would matter as we want the best first. then the `no_data_n` i presume gets always checked last anyway as this is a non-event with `AVALC` being set as `"N"`. by saying the event order is ignored is confusing to me as the `no_data_n` could be given equal weight then.